### PR TITLE
feat(z-image): add SGLang dedicated rollout engine support

### PR DESCRIPTION
**Pull request by xshrz**
## Summary

Adds Z-Image support to the SGLang dedicated rollout engine, as a follow-up
to the trainside integration (#41). The diffusion math already lives in the
SGLang fork's `ZImagePipeline`; this PR is the UniRL-side wiring:

- `_VALID_MODEL_FAMILIES` gains `"z_image"`.
- New recipe `z_image_sglang_replay_colocate.yaml` (colocate, LoRA sync,
  `old_logp_source: replay`, static shift=6.0, `FlowSDEStrategy`, CFG off).
- `patch_zimage_lora_pipeline.py` (+ hijack registration): defensive,
  idempotent `LoRAPipeline` mixin injection so `set_lora_from_tensors`
  works on stock-upstream wheels (no-op if already a `LoRAPipeline`).
- Conditions fix: emit `encoder_attention_mask` and length-gate it in the
  response translator — kept for single-encoder Z-Image (needed to trim
  variable-length Qwen3 captions in replay), dropped for SD3's multi-encoder
  fuse (mask_seq != embed_seq). SD3 behavior is unchanged by construction.
- Trainside recipe tuning: CFG off, `num_inference_steps=12`,
  `micro_batch_size=8`.

## Related Issue

N/A

## Test Plan

- `pre-commit run --files <changed>` — all hooks pass (ruff, check-yaml,
  recipe `_target_` resolution).
- `python scripts/check_recipe_targets.py` — all recipe `_target_` resolve.
- σ alignment: relies on `patch_set_timesteps` (neutralizes static + dynamic
  shift so the driver σ is used verbatim) with `verify_engine_used_sigmas`
  as the runtime guard.
- GPU smoke (2-card H20) — Not run yet; pending free hardware. Planned:
  `python -m unirl.train_diffusion --config-name=diffusion/z_image_sglang_replay_colocate +devices_per_node=2 num_devices=2 batch_size=2 sampling.samples_per_prompt=4 num_rollouts=2`
  watching `verify_engine_used_sigmas`, first `set_lora_from_tensors`, and
  the FlowGRPO importance ratio (~1 at iter 0).

## Compatibility / Risk

Additive: new model family + recipe + one defensive patch. The shared
`patch_conditions`/`response` change is SD3-safe by construction (length gate
reproduces SD3's existing mask drop). No change to existing models' behavior.

## Reviewer Notes

Opened as draft pending the GPU smoke. AI-assisted. Please sanity-check the
mask length-gate in `response._build_text_conditions` and the conditions
emit in `patch_conditions._POS_MAP`.

## Checklist

- [x] I reviewed the changed code and removed unrelated/generated artifacts.
- [x] I updated tests, docs, and configs where needed, or explained why not.

### CODE REVIEW

**Comment by xshrz** [comment]
Clipboard_Screenshot_1781504405
The yellow curve is the SGLang rollout engine, and the blue curve is TrainSide.

Command used:

```bash
bash examples/run_experiment_multinode_taiji.sh diffusion/z_image_sglang_full_tensor
```
